x-pack/plugins/security_solution/public/timelines/components/timeline/tour/index.tsx

```diff
@@ -86,21 +93,28 @@ const TimelineTourComp = () => {
 
   const isElementAtCurrentStepMounted = useIsElementMounted(nextEl);
 
+  const currentStepConfig = timelineTourSteps[tourState.currentTourStep - 1];
+
+  if (currentStepConfig?.timelineTab && currentStepConfig.timelineTab !== activeTab) {
+    switchToTab(currentStepConfig.timelineTab);
+  }
+
   if (!tourState.isTourActive || !isElementAtCurrentStepMounted) {
     return null;
   }
 
   return (
     <>
       {timelineTourSteps.map((steps, idx) => {
-        if (tourState.currentTourStep !== idx + 1) return null;
+        const step = idx + 1;
+        if (tourState.currentTourStep !== step) return null;
         return (
           <EuiTourStep
             panelProps={{
```

Comment:
nit, can you move the panelProps object out of the return below the `if`? Thanks!

Reply:
Done.